### ISSUE TITLE
task flow and prediction question

### ISSUE THREAD

**Issue by yddcode** [azavea/raster-vision#1306]
## ❓ Questions and Help

If you have a usage or general question, you may ask it here or in our [Gitter Channel](https://gitter.im/azavea/raster-vision). 

**Please check our [documentation](https://docs.rastervision.io/en/latest/index.html) first.**

If you want help with an error, please use the bug report template instead and fill in all the details.

Low-effort posts that are unclear, lack details, or are poorly formatted might be closed without a response.

---
I want to building segmentation, the task flow is chip classification first and then Semantic Segmentation?

I used the command 'python -m rastervision.pipeline.cli predict code/model-bundle.zip code/sample-img-spacenet-rio-cc.tif predict.tif', and got the figure below. Why?
![image](https://user-images.githubusercontent.com/54895981/139798914-a3fc0a61-6e31-4c48-b148-9c3050cbadcf.png)

**Comment by AdeelH** [comment]
>the task flow is chip classification first and then Semantic Segmentation?

No, those are separate tasks.

>I used the command 'python -m rastervision.pipeline.cli predict code/model-bundle.zip code/sample-img-spacenet-rio-cc.tif predict.tif', and got the figure below. Why?

The Spacenet Rio example is a chip classification example, so it outputs predictions as a GeoJSON rather than a TIFF file. I am assuming this is a screenshot of what you see when you open the predictions in QGIS. If so, you will need to modify the symbology settings for this layer in QGIS to color the chips according to the predicted class.

**Comment by yddcode** [comment]
Thanks.
What should I do when there are multiple images in the training set？

**Comment by AdeelH** [comment]
I don't understand the question. Please take a look at the examples here: https://github.com/azavea/raster-vision/tree/master/rastervision_pytorch_backend/rastervision/pytorch_backend/examples .

**Comment by yddcode** [comment]
I mean the example "{base_uri}/RGB-PanSharpen_AOI_2_Vegas_img205.tif" only one image, if i have 2 or more images, what should i do?

**Comment by AdeelH** [comment]
Ah, it seems like you're following the [tiny_spacenet](https://github.com/azavea/raster-vision/blob/master/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/tiny_spacenet.py) example (the one in the quickstart docs).

You will notice that on lines 50-55 https://github.com/azavea/raster-vision/blob/64f549d91d10d46d5a13f70b88ac3ab6b600c324/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/tiny_spacenet.py#L50-L55 we pass lists to the `train_scenes` and `validation_scenes` params. Since this is supposed to be a simple example, these lists have only one item each, but if you have multiple images then you can just add more items to these lists. For example, like this:

```python
 train_scenes=[ 
     make_scene('scene_1', train_image_uri1, train_label_uri1),
     make_scene('scene_2', train_image_uri2, train_label_uri2),
     make_scene('scene_3', train_image_uri3, train_label_uri3)
 ]
```

**Comment by yddcode** [comment]
thanks.
I used the 'rastervision predict output/bundle/modle-bundle.zip test.tif prediction', got the 'lables.tif'. Whers is the predict code, I want to make some changes due to the labels have some problems.

also, i want to the labels' building contours to convert the GeoJSON format, what should i do?
'是把标签转换为png然后提取轮廓还是利用QGIS还是有其他的方法？'
Is there another way to extract contours by converting labels to png or by using QGIS to do some processing? or other methods?

**Comment by AdeelH** [comment]
If I understand correctly, you want to get semantic segmentation predictions as a GeoJSON file. To do that, you have to specify the `vector_output` param when instantiating the `SemanticSegmentationLabelStoreConfig` object in your `get_config()` function. Example: https://github.com/azavea/raster-vision/blob/64f549d91d10d46d5a13f70b88ac3ab6b600c324/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/semantic_segmentation/spacenet_vegas.py#L114-L115 .

You will need to run the `bundle` command again after making this change. Then when you run the same `predict` command using the new model bundle you should see the GeoJSONs too.

>Whers is the predict code, I want to make some changes due to the labels have some problems.
The prediction code is a little spread out across multiple files. You might want to look at:
- rastervision_core/rastervision/core/predictor.py
- rastervision_pytorch_backend/rastervision/pytorch_backend/pytorch_semantic_segmentation.py
- rastervision_pytorch_learner/rastervision/pytorch_learner/semantic_segmentation_learner.py

**Comment by yddcode** [comment]
Aha, I have a GPU RTX3090, it is a great deal. But the deep learning enviornment CUDA CUDNN is doesn't work, how to deal with the question? I have try CUDA 11.3 torch=1.10 python=3.6

**Comment by AdeelH** [comment]
Please take a look at this thread and see if that works for you: https://gitter.im/azavea/raster-vision?at=611527769484630efa34a27d
